### PR TITLE
2.1 kvm race test fixes 1664437

### PR DESCRIPTION
**Pull request by jameinel**
This isn't critical for 2.1.0, as it is intended to be a code refactoring at testing change, but it does add tests to an area of the code that I've been working in that did not have much (if any). It also removes some of the code paths that were replaced, but left around because of what testing there *was* available.

## Description of change

PrepareHost wasn't tested directly, and even before my last refactoring, was only loosely tested. We can't test all of ContainerSetup because part of its function is to `apt install lxd` and actually deploy containers. So this is splitting off a portion of that into a helper object that can have its functionality overridden.

## QA steps

```go test```

Though if you want to make sure I haven't broken the functionality
```
  $ juju bootstrap maas
  $ juju deploy ubuntu --to lxd:0 -m controller
```

And see that the host networking is properly updated for the container.

## Documentation changes

Internal only refactoring.

## Bug reference

This was wanting to get the code tested as part of fixing [lp:1664437](https://pad.lv/1664437). It doesn't change the fix for it, but it does add tests that we are correctly interacting with a 'lock' object.

### CODE REVIEW

File: worker/provisioner/host_preparer_test.go [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+	}}
+	params := s.createPreparerParams(devices, nil)
+	lockName := params.LockName
+	ch := make(chan struct{})
+	close(ch)
+	params.AbortChan = ch
+	// This is what the AcquireLock should look like
+	params.AcquireLockFunc = func(abort <-chan struct{}) (mutex.Releaser, error) {
+		s.Stub.AddCall("AcquireLockFunc")
+		spec := mutex.Spec{
```
Comment (axw) [inline worker/provisioner/host_preparer_test.go]: Can we pare this back a bit please, so it's less integration and more unit test? We don't really care about mutexes specifically here. This should be sufficient to test the abort behaviour:

```go
params.AcquireLockFunc = func(abort <-chan struct{}) (mutex.Releaser, error) {
    select {
    case <-abort:
        return nil, errors.New("cancelled acquiring mutex")
    case <-time.After(coretesting.LongWait):
        return nil, errors.New("timed out waiting for mutex")
    }
}
```

Comment (jameinel) [inline worker/provisioner/host_preparer_test.go]: :+1: I also added a test that the abort channel that we set up is being properly passed into the AcquireLock.

File: worker/provisioner/host_preparer.go [hunk 2]
```diff
@@ -32,6 +41,7 @@ type HostPreparerParams struct {
 	CreateBridger      func() (network.Bridger, error)
 	AbortChan          <-chan struct{}
 	MachineTag         names.MachineTag
+	Logger			   loggo.Logger
```
Comment (axw) [inline worker/provisioner/host_preparer.go]: whitespace looks fishy (and in a couple of other spots), did you go fmt?

**Comment by jameinel** [comment]
I'm sure I missed it, but I'll catch it before landing. :) Thanks.

**Comment by jameinel** [comment]
$$merge$$

**Comment by jameinel** [comment]
$$merge$$ looks like Windows got very confused, will try again.

**Pull request merged**